core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java

```diff
@@ -105,39 +102,42 @@ public boolean receive(CarbonMessage carbonMessage, CarbonCallback carbonCallbac
                                 Response response) throws Exception {
         HttpUtil.setConnectionHeader(request, response);
-        PatternPathRouter.RoutableDestination<HttpResourceModel> destination =
-                currentMicroservicesRegistry.
-                        getMetadata().
-                        getDestinationMethod(request.getUri(), request.getHttpMethod(), request.getContentType(),
-                                request.getAcceptTypes());
+        PatternPathRouter.RoutableDestination<HttpResourceModel> destination = registry.getMetadata()
+                .getDestinationMethod(request.getUri(), request.getHttpMethod(), request.getContentType(),
+                        request.getAcceptTypes());
         HttpResourceModel resourceModel = destination.getDestination();
-        response.setMediaType(Util.getResponseType(request.getAcceptTypes(),
-                resourceModel.getProducesMediaTypes()));
-        InterceptorExecutor interceptorExecutor = new InterceptorExecutor(resourceModel, request, response,
-                                                                          currentMicroservicesRegistry
-                                                                                  .getInterceptors());
-        if (interceptorExecutor.execPreCalls()) { // preCalls can throw exceptions
-
-            HttpMethodInfoBuilder httpMethodInfoBuilder =
-                    new HttpMethodInfoBuilder().
-                            httpResourceModel(resourceModel).
-                            httpRequest(request).
-                            httpResponder(response).
-                            requestInfo(destination.getGroupNameValues());
-
-            HttpMethodInfo httpMethodInfo = httpMethodInfoBuilder.build();
-            if (httpMethodInfo.isStreamingSupported()) {
+        response.setMediaType(Util.getResponseType(request.getAcceptTypes(), resourceModel.getProducesMediaTypes()));
+        HttpMethodInfoBuilder httpMethodInfoBuilder = new HttpMethodInfoBuilder()
+                .httpResourceModel(resourceModel)
+                .httpRequest(request)
+                .httpResponder(response)
+                .requestInfo(destination.getGroupNameValues());
+        HttpMethodInfo httpMethodInfo = httpMethodInfoBuilder.build();
+        if (httpMethodInfo.isStreamingSupported()) {
+            Method method = resourceModel.getMethod();
+            Class<?> clazz = method.getDeclaringClass();
+            // Execute request interceptors
+            if (InterceptorExecutor.executeGlobalRequestInterceptors(registry, request, response)
+                    // Execute class level request interceptors
+                    && InterceptorExecutor.executeClassLevelRequestInterceptors(request, response, clazz)
+                    // Execute method level request interceptors
+                    && InterceptorExecutor.executeMethodLevelRequestInterceptors(request, response, method)) {
```

Comment:
It might be better to move this logic to a method to keep the dispatchMethod() clean.